packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -226,9 +228,26 @@ function hydrateStaticElement(elm: Node, vnode: VStatic, renderer: RendererAPI):
         return handleMismatch(elm, vnode, renderer);
     }
 
+    return hydrateStaticElementParts(elm, vnode, renderer);
+}
+
+function hydrateStaticElementParts(elm: Element, vnode: VStatic, renderer: RendererAPI) {
+    const { parts } = vnode;
+
+    if (!isUndefined(parts)) {
+        // Elements must first be set on the static part to validate against.
+        traverseAndSetElements(elm, parts, renderer);
```

Comment:
This step is needed before the validation of the static parts can occur since we compare vnodes against the actual element.

We could alternatively validate while we're traversing down the DOM nodes but I thought this was easier.